internal/daemonset/daemonset.go

```diff
@@ -82,6 +88,24 @@ func (d *daemonset) SetTopologyDaemonsetAsDesired(ctx context.Context, nfdInstan
 	return controllerutil.SetControllerReference(nfdInstance, topologyDS, d.scheme)
 }
 
+func (d *daemonset) DeleteDaemonSet(ctx context.Context, namespace, name string) error {
+	ds, err := d.getDaemonSet(ctx, namespace, name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get daemonset %s/%s: %w", namespace, name, err)
+	}
+
+	return d.client.Delete(ctx, ds)
```

Comment:
Hmm, this just looks a bit wonky and racy (theoretically, the check-then-act type). Would it be possible to do something like this (pseudo'ish golang example):

```suggestion
	err := xyz.Delete(ctx, ds)
	if k8serrors.IsNotFound(err) {
		return nil
	}
	return err
```

If not easy to do, let's go with what we have here

Reply:
Since the check-then-act is executed in the reconciliation loop, it should be ok even if the objects gets somehow deleted between "get" and "delete", since it will be certainly missing on the next loop before "get". But your suggestion is much cleaner , so i have fixed it :)